File: SakhaTyla.Core/Requests/WorkerRuns/Models/WorkerRunModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace SakhaTyla.Core.Requests.WorkerRuns.Models
{
    public class WorkerRunModel
    {
        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel ModificationUser { get; set; }

        [DisplayName("Worker")]
        public int WorkerInfoId { get; set; }
        public WorkerInfos.Models.WorkerInfoShortModel WorkerInfo { get; set; }

        [DisplayName("Status")]
        public Enums.WorkerRunStatus Status { get; set; }

        [DisplayName("Start Date")]
        public DateTime? StartDateTime { get; set; }

        [DisplayName("End Date")]
        public DateTime? EndDateTime { get; set; }

        [DisplayName("Data")]
        public string Data { get; set; }

        [DisplayName("Result")]
        public string Result { get; set; }

        [DisplayName("Result Data")]
        public string ResultData { get; set; }
    }
}

File: SakhaTyla.Web.Ng/src/app/worker-run-core/worker-run.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { WorkerRunFilter } from './worker-run-filter.model';
import { WorkerRunStatus } from '../worker-run-status/worker-run-status.model';

export class WorkerRun {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  workerInfoId: number;
  status: WorkerRunStatus;
  startDateTime: Date;
  endDateTime: Date;
  data: string;
  result: string;
  resultData: string;
}

export class WorkerRunListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new WorkerRunFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}
